### PR TITLE
compile btn logic

### PR DESCRIPTION
**Pull request by alisya-k**
- added POST form and event handler to reach /compile in backend

### CODE REVIEW

File: lombardy/templates/index.html [hunk 1]
```diff
@@ -14,18 +14,25 @@
...
   <body>
     <h1>Epic Compiler 'Splorer</h1>
-    <div id="editor1">
+    <div name="editor1" id="editor1">
       function foo(items) { var x = "All asdflkasjdflk is syntax highlighted";
       return x; }
     </div>
+    <br />
+
+    <form action="/compile" method="post" id="compileForm">
```
Comment (aidenfoxivey) [inline lombardy/templates/index.html:23]: Is a form necessary if you're reading off the DOM?

Should be able to do an event listener 
`document.getElementById("myBtn").addEventListener("click", displayDate);` that just takes the current value

editor.getValue()

and then writes a POST through `fetch` to the backend with the data

File: lombardy/src/main.go [hunk 2]
```diff
@@ -54,6 +47,19 @@ func main() {
 
 		// objdump -S a.out
 
+		c.HTML(http.StatusOK, "index.html", gin.H{
```
Comment (aidenfoxivey) [inline lombardy/src/main.go:50]: why is compile returning index.html

**Comment by aidenfoxivey** [comment]
Thoughts on a refactor that removes the Post form and does some JS instead?

File: lombardy/templates/index.html (hunk 1 above)
Comment (alisya-k) [reply lombardy/templates/index.html:23]: I agree I actually forgot that existed

File: lombardy/src/main.go (hunk 2 above)
Comment (alisya-k) [reply lombardy/src/main.go:50]: It should redirect to it with data, I did not know how to do that lol

File: lombardy/src/main.go [hunk 3]
```diff
@@ -3,7 +3,7 @@ package main
 import (
 	"fmt"
 	"net/http"
-	"os/exec"
+	// "os/exec"
```
Comment (aidenfoxivey) [inline lombardy/src/main.go:6]: If you need to remove this, then probably just remove this - otherwise uncomment

File: lombardy/src/main.go [hunk 4]
```diff
@@ -29,34 +29,27 @@ func main() {
...
+			c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+			return
+		}
 
 		// TODO: get compiled code here
-		cmd := exec.Command("cc", "main.c")
-		if err := cmd.Run(); err != nil {
-			panic(err)
-		}
+		// cmd := exec.Command("cc", "main.c")
```
Comment (aidenfoxivey) [inline lombardy/src/main.go:40]: Why are we commenting this out here?

File: lombardy/src/main.go [hunk 5]
```diff
@@ -29,34 +29,27 @@ func main() {
...
-			"editor2text": compileText,
-		})
-
-		// if err := c.BindJSON(&src); err != nil {
-		// 	c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
-		// 	return
-		// }
-
-		//c.JSON(http.StatusOK, gin.H{})
+		c.JSON(http.StatusOK, gin.H{"res": src.Source})
```
Comment (aidenfoxivey) [inline lombardy/src/main.go:52]: I take it this is just echoing then.

**Review by aidenfoxivey — CHANGES_REQUESTED** [review]

File: lombardy/src/main.go (hunk 5 above)
Comment (alisya-k) [reply lombardy/src/main.go:52]: yes it is a lovely poc

File: lombardy/src/main.go (hunk 3 above)
Comment (alisya-k) [reply lombardy/src/main.go:6]: it's removed in the cosmic sense

**Review by aidenfoxivey — APPROVED** [review]

**Pull request merged**